### PR TITLE
close Knawat/dropshipping-woocommerce#102

### PR DESCRIPTION
**Pull request by hamusthekiller**
close Knawat/dropshipping-woocommerce#102
Update Fix for issue #102
Appmaker.xyz mobile application crash fix when sending orders to knawat.com

### CODE REVIEW

File: includes/class-dropshipping-woocommerce-orders.php [hunk 1]
```diff
@@ -1230,7 +1230,9 @@ function knawat_dropshipwc_add_suborders_api( $response, $post, $request ){
 		}
 
         $suborder_ids = array();
-        if( 0 == $post->get_parent_id() ){
+        
+        //manual fix for $post->get_parent_id() to check if it exists in (04/10/2019)
```
Comment (ElGreetly) [inline includes/class-dropshipping-woocommerce-orders.php]: this just lining the '//' should be followed by space I don't know why there a date here?

File: includes/class-dropshipping-woocommerce-orders.php [hunk 2]
```diff
@@ -1230,7 +1230,9 @@ function knawat_dropshipwc_add_suborders_api( $response, $post, $request ){
...
+        if( $post && method_exists($post, 'get_parent_id') && 0 == $post->get_parent_id() ){
```
Comment (ElGreetly) [inline includes/class-dropshipping-woocommerce-orders.php]: this snippet `0 == $post->get_parent_id()` should be `$post->get_parent_id() == 0`

File: includes/class-dropshipping-woocommerce-orders.php [hunk 3]
```diff
@@ -1257,7 +1259,9 @@ function knawat_dropshipwc_add_localds_api( $response, $post, $request ){
             return $response;
         }
 
-        $knawat_order_ds = get_post_meta( $post->get_id(), '_knawat_order_ds', true );
+        //manual fix for $post->get_id() to check if it exists in (4/10/2019 , 06:35)
```
Comment (ElGreetly) [inline includes/class-dropshipping-woocommerce-orders.php]: I don't understand about the time on the comment pls explain

File: includes/class-dropshipping-woocommerce-orders.php [hunk 4]
```diff
@@ -1257,7 +1259,9 @@ function knawat_dropshipwc_add_localds_api( $response, $post, $request ){
...
+        if (method_exists($post,'get_id')){
+            $knawat_order_ds = get_post_meta( $post->get_id(), '_knawat_order_ds', true );}
```
Comment (ElGreetly) [inline includes/class-dropshipping-woocommerce-orders.php]: I thought that the please add one line for the `}`

File: includes/class-dropshipping-woocommerce-orders.php (hunk 2 above)
Comment (ElGreetly) [reply includes/class-dropshipping-woocommerce-orders.php]: I also want to know if you are checking the existence of `$post` or if the value is truthy or falsy

File: includes/class-dropshipping-woocommerce-orders.php (hunk 1 above)
Comment (hamusthekiller) [reply includes/class-dropshipping-woocommerce-orders.php]: @ElGreetly The time is for me to keep tracking just my own sample of tracking issues.

File: includes/class-dropshipping-woocommerce-orders.php (hunk 2 above)
Comment (hamusthekiller) [reply includes/class-dropshipping-woocommerce-orders.php]: @ElGreetly the snippet is part of the main code and it is crucial for the plugin to work. It compares the false with the value it got from API that was stored in post and it executes faster.

File: includes/class-dropshipping-woocommerce-orders.php (hunk 3 above)
Comment (hamusthekiller) [reply includes/class-dropshipping-woocommerce-orders.php]: @ElGreetly my own tracking for issues solved.

File: includes/class-dropshipping-woocommerce-orders.php (hunk 2 above)
Comment (ElGreetly) [reply includes/class-dropshipping-woocommerce-orders.php]: for this `$post->get_parent_id() == 0` I am talking about linting or convention to but the static value at the right side it has no issue with the faster execution 
for the `$post` falsy truthy check I didn't understand the reason.

File: includes/class-dropshipping-woocommerce-orders.php [hunk 5]
```diff
@@ -1230,7 +1230,9 @@ function knawat_dropshipwc_add_suborders_api( $response, $post, $request ){
 		}
 
         $suborder_ids = array();
-        if( 0 == $post->get_parent_id() ){
+        
+        // manual fix for $post->get_parent_id() to check if it exists in (04/10/2019)
```
Comment (ElGreetly) [inline includes/class-dropshipping-woocommerce-orders.php]: putting the date on the comments is not accepted the comment must indicate some information about the code if you want to track the code dates you can check git and commits

File: includes/class-dropshipping-woocommerce-orders.php [hunk 6]
```diff
@@ -1257,7 +1259,10 @@ function knawat_dropshipwc_add_localds_api( $response, $post, $request ){
             return $response;
         }
 
-        $knawat_order_ds = get_post_meta( $post->get_id(), '_knawat_order_ds', true );
+        // manual fix for $post->get_id() to check if it exists in (4/10/2019 , 06:35)
+        if (method_exists($post,'get_id')){
+            $knawat_order_ds = get_post_meta( $post->get_id(), '_knawat_order_ds', true );
+	}
```
Comment (ElGreetly) [inline includes/class-dropshipping-woocommerce-orders.php]: Please, the be give the linting some `}` indentation should by aligned under the `if`

**Pull request merged**